src/sentry/api/endpoints/organization_events_meta.py

```diff
@@ -88,6 +88,94 @@ def get(self, request, organization):
         return Response(baseline_data)
 
 
+class OrganizationEventsTrace(OrganizationEventsEndpointBase):
+    def get(self, request, organization, trace_id):
+        if not self.has_feature(organization, request):
+            return Response(status=404)
+
+        try:
+            params = self.get_snuba_params(request, organization)
+        except NoProjects:
+            return Response(status=404)
+
+        # Get 100 events from snuba, and pray these cover the trace
+        with self.handle_query_errors():
+            result = discover.query(
+                selected_columns=[
+                    "id",
+                    "timestamp",
+                    "transaction",
+                    "project_id",
+                    "trace.span",
+                    "trace.parent_span",
+                ],
+                orderby=["-timestamp", "trace.parent_span", "id"],
+                params=params,
+                query=f"event.type:transaction trace:{trace_id}",
+                limit=100,
+                referrer="experimental.api.trace-view.get_ids",
+            )
+            if len(result["data"]) == 0:
+                return Response(status=404)
+
+        # Get all our events
+        events = {
+            event.event_id: event
+            for event in eventstore.get_events(
+                eventstore.Filter(
+                    project_ids=params["project_id"],
+                    event_ids=[event["id"] for event in result["data"]],
+                )
+            )
+        }
+
+        with sentry_sdk.start_span(op="parsing snuba response"):
```

Comment:
```suggestion
        with sentry_sdk.start_span(op="building trace tree"):
```

The snuba response has already been parsed and turned into python objects at this point.